### PR TITLE
AL OMAS, Open Lineage: Storing Lineage

### PR DESCRIPTION
**Pull request by DimitriosMaimaris**
Functionality for saving Entities and Relationships relating to Horizontal Lineage in bufferGraph.

### CODE REVIEW

File: open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/pom.xml [hunk 1]
```diff
@@ -27,6 +27,13 @@
     </properties>
     <dependencies>
 
+
+        <dependency>
+            <groupId>org.quartz-scheduler</groupId>
```
Comment (planetf1) [inline open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/pom.xml]: This is introducing a significant new dependency. Can this be brought up on the next Egeria call to ensure the core maintainers are happy with the additional dependencyu?

File: open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/pom.xml [hunk 2]
```diff
@@ -27,6 +27,13 @@
...
+            <artifactId>quartz</artifactId>
+            <version>2.2.1</version>
```
Comment (planetf1) [inline open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/pom.xml]: Current version is 2.3.1 . Was there a reason the older version was chosen?

Comment (planetf1) [inline open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/pom.xml]: Also the group/artifact/version should be added into the '<dependencyManagement>' section in the top level pom.xml, then here we remove the version. This helps to maintain consistency and is documented in the developer guidelines

File: open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/pom.xml [hunk 3]
```diff
@@ -78,6 +85,13 @@
             <artifactId>gremlin-core</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.lucene</groupId>
+            <artifactId>lucene-analyzers-common</artifactId>
+            <version>7.0.0</version>
```
Comment (planetf1) [inline open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/pom.xml]: Current versions are 8.2.x. However we probably need to be consistent with the rest of egeria (and janus) if possible.  See also #1106 cc: @grahamwallis 

If a new dependency is needed - as with previous comment, please ensure it is added to dependencyManagement and the explicit version removed here. Also I would recommend running 'mvn dependency:tree' in this case, as we have transient dependencies coming in from Janus. (let me know if you need any help)

File: open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/mockdata/MockGraphGenerator.java [hunk 4]
```diff
@@ -41,8 +41,8 @@ public MockGraphGenerator() {
     /**
      * The parameters for the graph that is to be generated are hardcoded for now.
      * A "flow" constitutes of a path between columns of different tables, connected together by process nodes.
```
Comment (GuyAckermans) [inline open-metadata-implementation/governance-servers/open-lineage-services/open-lineage-services-server/src/main/java/org/odpi/openmetadata/governanceservers/openlineage/mockdata/MockGraphGenerator.java]: @code{} should remain

**Comment by planetf1** [comment]
Just a FYI, JavaDoc is now run on every build. Whilst we don't check that all methods are documented (we could!), it is important that the javadoc copiler doesn't find errors. The latest PR build seems to be failing on the javadoc in the lineage modules. You can run mvn javadoc:javadoc just to run this stage.

**Pull request merged**